validator/client/validator.go

```diff
@@ -131,6 +134,86 @@ func (v *validator) Done() {
 	v.ticker.Done()
 }
 
+func (v *validator) Init(ctx context.Context) error {
+	ctx, span := trace.StartSpan(ctx, "validator.Init")
+	defer span.End()
+
+	ticker := time.NewTicker(backOffPeriod)
+	defer ticker.Stop()
+
+	firstTime := true
+	var (
+		currentSlot primitives.Slot
+		err         error
+	)
+	for {
+		if !firstTime {
+			if ctx.Err() != nil {
+				log.Info("Context canceled, stopping validator")
+				return errors.New("context canceled")
+			}
+			<-ticker.C
+		}
+
+		firstTime = false
+		currentSlot, err = v.WaitForChainStart(ctx)
+		if err != nil {
+			if isConnectionError(err) {
+				log.WithError(err).Warn("Could not determine if beacon chain started")
+				continue
+			}
+
+			return errors.Wrap(err, "Could not determine if beacon chain started")
+		}
+
+		if err := v.WaitForKeymanagerInitialization(ctx); err != nil {
+			return errors.Wrap(err, "Wallet is not ready")
+		}
+
+		if err := v.WaitForSync(ctx); err != nil {
+			if isConnectionError(err) {
+				log.WithError(err).Warn("Could not determine if beacon chain started")
+				continue
+			}
+
+			return errors.Wrap(err, "Could not determine if beacon node synced")
+		}
+
+		if err := v.WaitForActivation(ctx, nil /* accountsChangedChan */); err != nil {
+			return errors.Wrap(err, "Could not wait for validator activation")
+		}
+
+		if err := v.CheckDoppelGanger(ctx); err != nil {
+			if isConnectionError(err) {
+				log.WithError(err).Warn("Could not wait for checking doppelganger")
+				continue
+			}
+
+			return errors.Wrap(err, "Could not succeed with doppelganger check")
+		}
+		break
+	}
+
+	if err := v.UpdateDuties(ctx, currentSlot); err != nil {
+		handleAssignmentError(err, currentSlot)
+	}
+
+	km, err := v.Keymanager()
+	if err != nil {
+		return errors.Wrap(err, "Could not get keymanager")
+	}
```

Comment:
In the beginning of `run` we have
```
km, err := v.Keymanager()
if err != nil {
	log.WithError(err).Fatal("Could not get keymanager")
}
```
Because of `Fatal` we stop the node when we can't get the keymanager. So firstly, as I noted in another comment, we can't ignore this error. And secondly, we should put this check and `WaitForKeymanagerInitialization` before `WaitForChainStart` because these are very quick checks and will fail fast.

Reply:
we don't ignore this error, it's triggered in line 43 in the runner